File: week_1/day-2-prefix-sum-hashing/subarray_sum_k_follow_up_questions/subarray_sum_positive.py
# subarray_sum_positive.py

def subarraySumPositive(nums, k):
    left = 0
    curr_sum = 0
    count = 0

    for right in range(len(nums)):
        curr_sum += nums[right]

        while curr_sum > k:
            curr_sum -= nums[left]
            left += 1

        if curr_sum == k:
            count += 1

    return count

# Example Input
nums = [1, 2, 1, 1, 1]
k = 3

# Function Call
print(subarraySumPositive(nums, k))  # Output: 4

File: week_1/day-2-prefix-sum-hashing/subarray_sum_k_follow_up_questions/subarray_sum_positive.js
// subarray_sum_positive.js

function subarraySumPositive(nums, k) {
    let left = 0;
    let currSum = 0;
    let count = 0;

    for (let right = 0; right < nums.length; right++) {
        currSum += nums[right];

        while (currSum > k) {
            currSum -= nums[left];
            left++;
        }

        if (currSum === k) {
            count++;
        }
    }

    return count;
}

// Example Input
const nums = [1, 2, 1, 1, 1];
const k = 3;

// Function Call
console.log(subarraySumPositive(nums, k));  // Output: 4
